src/Installer/Installer/Factory.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Dock\Installer\Installer;
+
+use Dock\System\OS;
+
+class Factory
+{
+    public function getInstaller(OS $os)
+    {
+        $operatingSystem = $os->get();
+
+        switch ($operatingSystem) {
```

Comment:
I think it's more readable. And not a lot of other operating system :stuck_out_tongue: